python/test/unit/language/test_core.py

```diff
@@ -2139,6 +2134,11 @@ def kernel(X, Z, TO_TYPE: tl.constexpr, BITCAST: tl.constexpr, SIZE: tl.constexp
         else:
             torch.testing.assert_close(z_ref, z_tri, rtol=0, atol=0)
     else:
+        # Triton clamps negative values to zero (when casting float numbers
+        # to unsigned integers), while numpy wraps around intmax.
+        # Triton version is selected as the expected behavior.
+        if dtype_z in uint_dtypes and 'float' in dtype_x:
+            x = np.where(x >= 0, x, 0)
```

Comment:
You're still changing `x` not `z_ref`?

Reply:
It should be safe now (I moved it after kernel execution and it should not affect `z_tri`). After the conversion it will be impossible to clearly understand what needs to be changed (in `z_ref`). Okay, maybe it's possible knowing the initial value boundaries, but it's more complicated.